0000-settlement-engines/0000-settlement-engines.md

```diff
@@ -0,0 +1,491 @@
+  - Positive amount indicates its peer is indebted to the connector (an _asset_ to the connector).
+  - Negative amount indicates its peer has sent a pre-payment to the connector.
+
+(Note that "accounts payable" and "accounts receivable" refer to accounts as _records_, distinct from accounts referring to the _arrangement_ between counterparties.)
+
+Thus, the connector's accounts payable with its peer should mirror its peer's accounts receivable with the connector, and respectively, the connector's accounts receivable should equal its peer's accounts payable.
+
+### Settlement
+
+A **settlement** is the irrevocable discharge of a liability by providing something of value to the party to whom the liability is owed.
+
+Settlements may occur on a **settlement system**, or medium for exchanging value. Some settlements may transfer funds on a **ledger**, or registry of account balances and/or transactions, which is a type of settlement system. (Although not all settlement systems are ledgers, here, the terms are sometimes used interchangeably.)
+
+Examples of settlement systems include:
+
+- Bank clearing houses
+- Credit card processors
+- Money transfer services
+- Cryptocurrencies, blockchains, and distributed ledgers
+- Payment channels and layer 2 networks
+- Cash or physical delivery of assets
+
+The accounting system is responsible for triggering outgoing settlements. For example, when the accounts payable reaches a particular threshold, the accounting system could trigger a settlement to reduce the liabilities owed to the peer to a predefined, lesser amount.
+
+### Double-entry bookkeeping
+
+The interface in this specification provides a mechanism to reconcile settled cash balances, representing _value_, with the aforementioned account balances tracked by the accounting system, representing _credit_.
+
+Together, a settlement engine and an accounting system interface with one another to perform double-entry bookkeeping.
+
+To ensure accurate, balanced double-entry bookkeeping, settlement engine and accounting system implementations MUST enforce several invariants.
+
+#### Settlement engine invariants
+
+##### Settlement symmetry
+
+Individual settlement engine implementations define how value is transferred to the counterparty: this specification merely defines settlements in terms of how settlement engines and accounting systems must behave.
+
+The fundamental expected behavior of a settlement engine implementation is the sum of amounts one instance is instructed to settle eventually equals the sum of amounts the recipient instance instructs its accounting system to credit as incoming settlements.
+
+Many factors may cause temporary or lasting inconsistencies between these instructed and acknowledged settlements. For example:
+
+- External conditions, such as the time to finalize settlements on an underlying ledger
+- Unintended bugs in the settlement engine implementation
```

Comment:
@sappenin and I had a long back and forth related to this. It came down, it's weird to say only "settlement engines MUST ensure that the the instructed settlements are eventually acknowledged by the peer" since both the implementation and/or many external factors could prevent that from happening -- even though this is fundamental to the spec.

I removed that "unintended bugs" reference, and updated this section to categorize and clarify the different reasons for the inconsistencies. Broadly, it's: settlement delay, failed outgoing settlements, uncredited incoming settlements, and operating factors (such as if peers are operating the wrong settlement engine) that can result in unbalanced double entry.